Data/Vector/Generic.hs

```diff
@@ -1822,18 +1822,22 @@ scanr' f z = unstreamR . Bundle.scanl' (flip f) z . streamR
 -- | /O(n)/ Right-to-left scan over a vector with its index
 iscanr :: (Vector v a, Vector v b) => (Int -> a -> b -> b) -> b -> v a -> v b
 {-# INLINE iscanr #-}
-iscanr f z =
+iscanr f z v =
     unstreamR
```

Comment:
I thought vector stream fusion can support an exact size? Using the Exact
size constructor

Hrmmmm. Maybe this code should use that?

On Sunday, July 24, 2016, dolio <notifications@github.com> wrote:

> In Data/Vector/Generic.hs
> <https://github.com/haskell/vector/pull/125#discussion_r71997772>:
>
> > @@ -1822,18 +1822,22 @@ scanr' f z = unstreamR . Bundle.scanl' (flip f) z . streamR
> >  -- | /O(n)/ Right-to-left scan over a vector with its index
> >  iscanr :: (Vector v a, Vector v b) => (Int -> a -> b -> b) -> b -> v a -> v b
> >  {-# INLINE iscanr #-}
> > -iscanr f z =
> > +iscanr f z v =
> >      unstreamR
>
> I think that's new with the bundle stuff. Unfortunately this function
> probably breaks fusion, because of the need for length information.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/haskell/vector/pull/125/files/02d4dc8339c0d607c9bcf5649cb750a5dec2b941#r71997772>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAQwrD1FDB8pKWfd-m1RTA7sdpB_UGbks5qY_NDgaJpZM4JTo-3>
> .
>